dev/archery/archery/integration/datagen.py

```diff
@@ -193,6 +193,28 @@ def generate_range(self, size, lower, upper, name=None,
         return PrimitiveColumn(name, size, is_valid, values)
 
 
+# Integer field that fulfils the requirements for the run ends field of RLE.
+# The integers are positive and in a strictly increasing sequence
+class RunEndsField(IntegerField):
+    def __init__(self, name, bit_width, *, nullable=False,
+                 metadata=None):
+        super().__init__(name, is_signed=True, bit_width=bit_width,
+                         nullable=nullable, metadata=metadata, min_value=1)
+
+    def generate_range(self, size, lower, upper, name=None,
+                       include_extremes=False):
+        # values = np.random.randint(lower, upper, size=size, dtype=np.int64)
+        rng = np.random.default_rng()
+        values = rng.choice(2 ** (self.bit_width - 1) - 1, size=size, replace=False)
```

Comment:
it's perfectly legal to start run-ends on a non-zero value because they represent the *ends* of a run as opposed to the way offsets work where you have a start and end. So if you have an REE array to represent `[3, 3, 3, 4, 4, 4]` you have two run_ends: `[3, 6]`.  As the run-end should be the index of the start of the next run.

While it's technically not *invalid*, it wouldn't make any sense for run-ends to start on a 0 value honestly.

Reply:
A way of interpreting run-ends that is useful to me in understanding them:

`run_end = base_index + length` and `length > 0` because an empty run (`length == 0`) doesn't have to exist.

Reply:
This also matches C++ definitions of iterators: `end() == begin() + size()`.

Reply:
```suggestion
        values = rng.choice(2 ** (self.bit_width - 1) - 1, size=size, replace=False)
        values += 1
```
> A run must have have a length of at least 1. This means the values in the
run ends array all are positive and in strictly ascending order.
In the spec, we explicitly forbid zero-length runs. I think what is missing here is increasing each array value by 1. The range end argument passed to rng.choice already contains `- 1`. Since the argument to rng.choice is defined as the first value that is not possible, the largest value we generate currently is `2 ** (self.bit_width - 1) - 2`, which seems one to low